test/extensions/filters/network/thrift_proxy/buffer_helper_test.cc

```diff
@@ -605,7 +607,7 @@ TEST(BufferHelperTest, WriteDouble) {
 
   {
     Buffer::OwnedImpl buffer;
-    BufferHelper::writeDouble(buffer, -DBL_MAX);
+    BufferHelper::writeDouble(buffer, std::numeric_limits<double>::lowest());
```

Comment:
There are static_asserts in the Thrift code that require ISO IEC-559 (aka IEEE-754) doubles of width 8 bytes, so if ever Envoy were ported to platform that didn't support that standard or used a different size for double, we'd get a compile error and some work would need to be done. That said, I'm not aware of any platform where this would fail.

FWIW, the protobuf wire format seems to make the same assumption about the width of doubles (https://github.com/google/protobuf/blob/master/src/google/protobuf/wire_format_lite.h#L818) and that the double format on all supported platforms is the same. (Though there may be something elsewhere in the build that guarantees that code will work.)